server/src/tables.rs

```diff
@@ -40,240 +47,361 @@ use datafusion::physical_plan::{
             model_table_metadata: model_table_metadata.clone(),
-            segment_folder_path: Path::from(model_table_metadata.segment_path.clone()),
             object_store_url: ObjectStoreUrl::local_filesystem(),
             schema: Arc::new(Schema::new(columns)),
+            fallback_field_column: fallback_field_column as u64,
         })
     }
 
-    fn rewrite_and_combine_filters(&self, filters: &[Expr]) -> Option<Expr> {
-        // TODO: implement rewriting of members to group ids.
-        let rewritten_filters: Vec<Expr> = filters
-            .iter()
-            .map(|filter| match filter {
-                Expr::BinaryExpr { left, op, right } => {
-                    if **left == col("tid") {
-                        // Assumes time series are not grouped so tids and gids are equivalent.
-                        self.binary_expr(col("gid"), *op, *right.clone())
-                    } else if **left == col("timestamp") {
-                        match op {
-                            Operator::Gt => {
-                                self.binary_expr(col("end_time"), *op, self.to_i64(right))
-                            }
-                            Operator::GtEq => {
-                                self.binary_expr(col("end_time"), *op, self.to_i64(right))
-                            }
-                            Operator::Lt => {
-                                self.binary_expr(col("start_time"), *op, self.to_i64(right))
-                            }
-                            Operator::LtEq => {
-                                self.binary_expr(col("start_time"), *op, self.to_i64(right))
-                            }
-                            Operator::Eq => self.binary_expr(
-                                self.binary_expr(
-                                    col("start_time"),
-                                    Operator::LtEq,
-                                    self.to_i64(right),
-                                ),
-                                Operator::And,
-                                self.binary_expr(
-                                    col("end_time"),
-                                    Operator::GtEq,
-                                    self.to_i64(right),
-                                ),
-                            ),
-                            _ => filter.clone(),
-                        }
-                    } else {
-                        filter.clone()
-                    }
-                }
-                _ => filter.clone(),
-            })
-            .collect();
-        combine_filters(&rewritten_filters)
+    // TODO: Move to the metadata component when it exists.
+    /// Compute the 64-bit keys of the univariate time series to retrieve from
+    /// the storage engine using the fields, tag, and tag values in the query.
+    /// Returns a [`DataFusionError::Plan`] if the necessary data cannot be
+    /// retrieved from [`catalog::METADATA_SQLITE_NAME`].
```

Comment:
Sometimes we just call it the `metadata database`. Is [`catalog::METADATA_SQLITE_NAME`] the preferred way to refer to the metadata SQLite database?

Reply:
I have changed it to say `metadata database` everywhere as it required the fewest number of changes and will still be correct if another RDBMS is used for metadata.